### PR TITLE
Add Utilities.sorted_list_of_model_attributes

### LINKED ISSUE #2366: Order of model attributes during copy
As reported in https://github.com/jump-dev/MathOptInterface.jl/issues/2363#issuecomment-1862620395, the `SlackBridgePrimalDualStart` is ignored if it is copied before the objective is set which depends on the order of the model attributes.

### PR DESCRIPTION
**Pull request by odow**
Closes #2366
Closes #2369 

https://github.com/jump-dev/MathOptInterface.jl/actions/runs/7282138662

### CODE REVIEW

File: src/Utilities/copy.jl [hunk 1]
```diff
@@ -27,12 +59,7 @@ function pass_attributes(
     src::MOI.ModelLike,
     index_map::IndexMap,
 )
-    attrs = MOI.get(src, MOI.ListOfModelAttributesSet())
-    # We need to deal with the UserDefinedFunctions first, so that they are in
-    # the model before we deal with the objective function or the constraints.
-    # We also need `ObjectiveSense` to be set before `ObjectiveFunction`.
-    sort!(attrs; by = _sort_priority)
-    for attr in attrs
+    for attr in MOI.get(src, MOI.SortedListOfModelAttributesSet())
```
Comment (odow) [inline src/Utilities/copy.jl:62]:
```suggestion
    for attr in MOI.get(src, SortedListOfModelAttributesSet())
```

**Comment by odow** [comment]
Looking into why Juniper started to fail tests

**Comment by odow** [comment]
So this issue is that Juniper defines a bunch of model filters:
https://github.com/lanl-ansi/Juniper.jl/blob/05ae25efee510ef6d4cbe90bb098460ec728a8f3/src/filter.jl#L1-L29
and we can't define a new `AbstractModelAttribute` like this without breaking existing behavior.

File: src/Utilities/copy.jl [hunk 2]
```diff
@@ -9,9 +9,39 @@
 
 include("copy/index_map.jl")
 
-_sort_priority(::Any) = 2
-_sort_priority(::MOI.UserDefinedFunction) = 0
-_sort_priority(::MOI.ObjectiveSense) = 1
```
Comment (odow) [inline src/Utilities/copy.jl:14]: So in retrospect, all this PR really needed was `_sort_priority(::Any) = 3` and `_sort_priority(::MOI.ObjectiveFunction) = 2`.

So we want to make this all public?

**Comment by odow** [comment]
https://github.com/jump-dev/MathOptInterface.jl/actions/runs/7282725778

File: src/Utilities/copy.jl (hunk 2 above)
Comment (joaquimg) [reply src/Utilities/copy.jl:14]: I would do this simple fix, and reopen this PR if this keeps coming back and someone actually needs to extend this kind of methods.

**Comment by odow** [comment]
Closing in favor of #2372

**Pull request closed without merging**